From 0c86790d67fecd40d43e4b5edb46760f6bb4e949 Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Tue, 3 Jul 2018 21:00:13 -0500
Subject: [PATCH 1775/2367] ASoC: SOF: ops: fix uninitialized use warning

For some architectures/compilers, 0-day reports errors

sound/soc/sof/ops.c: In function 'snd_sof_pci_update_bits_unlocked':
sound/soc/sof/ops.c:26:6: warning: 'ret' is used uninitialized in this function [-Wuninitialized]
  u32 ret;
      ^~~

Fix by forcing value to ~0 (as done in pci_read_config_dword)

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
(cherry picked from commit 28fc73b252eaa6438e9c10a854f7067db24f07cd)
---
 sound/soc/sof/ops.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/sof/ops.c b/sound/soc/sof/ops.c
index a3b1aeb..0668f64 100644
--- a/sound/soc/sof/ops.c
+++ b/sound/soc/sof/ops.c
@@ -23,7 +23,7 @@ int snd_sof_pci_update_bits_unlocked(struct snd_sof_dev *sdev, u32 offset,
 {
 	bool change;
 	unsigned int old, new;
-	u32 ret;
+	u32 ret = ~0; /* explicit init to remove uninitialized use warnings */
 
 	pci_read_config_dword(sdev->pci, offset, &ret);
 	dev_dbg(sdev->dev, "Debug PCIR: %8.8x at  %8.8x\n",
-- 
2.7.4

